### ISSUE TITLE
Overwriting sessions

### ISSUE THREAD

**Issue by miraodasilva** [wandb/wandb#1450]
Hi,

I would like to experiment with my code, running it multiple times in a row while debugging/tuning it  without creating a single session for every run, instead having one session that keeps getting overwritten every time, to avoid spamming the project with lots of sessions. 

I used to do this on visdom easily by always using the same name for my env and clearing it in the code automatically. Is there a way to do this on wandb? From what I can see deleting runs from the code is possible but it is said in the docs you can't use the ID from the deleted session, so I don't really see how I could do what I describe above.

Thanks a lot.

**Comment by vanpelt** [comment]
Hey @miraodasilva we're working on a "disabled" mode that can be used for debugging scenarios where you don't actually want to create a new run in wandb each time.  That should be released later this week.  If you really do want to create a run in wandb during debugging, I suggest setting a unique id via an environment variable and using it in your calls to wandb.init.  i.e.

```
import wandb
print("Add the following to your environment to have the same run used between executions:")
print("WANDB_RUN_ID="+wandb.util.generate_id())
```

**Comment by miraodasilva** [comment]
I see, does this mean that I should use the same ID every time? If so, I am fairly confident that this does not overwite the session correctly. If I use the same ID nothing is logged to the previously created session with the same ID. Ideally, I would like to clear it and see the new datapoints. Thanks a lot for your time.

**Comment by vanpelt** [comment]
Specifying the the RUN_ID will use the existing ID and not create a new "run" in the system.  Do you have an example of this not happening?

**Comment by miraodasilva** [comment]
Indeed, a new run is not created but I cannot visualize any of the new data points on the plots. I apologize if I am not being clear in my explanation. 

Basically, code run A creates wandb run with ID 1 and I can visualize the new loss in the plots as it is updated in the code. When I run it again, I start code run B with the same ID (1). No new wandb run is created, but in run 1 I can only see the loss from code run A, not from code run B, which is what I would like to see now, since B should have overwritten A.

Thanks.

**Comment by vanpelt** [comment]
Ah I see, it's because of the `step` counter in your log calls.  What you actually want to do here is set `resume="allow"` and then we'll append any new metrics to the old run.  We don't support overwriting existing metrics currently, only appending.  In a future release we may support overwriting old data.

**Comment by miraodasilva** [comment]
I see, thanks a lot for your answer. As I thought, there does not appear to be a way to do this. There is no way to delete runs from API yet, right? I saw an issue about it but it never gained any traction from what I can gather https://github.com/wandb/client/issues/702 . If there was a way to do this, I could just use the env var as a counter, which would keep track of the ID of the previous run, delete it and then set the env var to a new value. This logic could be hidden away in a function and make things quite simple.

**Comment by cvphelps** [comment]
Hi there, we don't support this right now but you can filter in the table in the UI and get all the runs and remove them at once with "select all" in a bulk delete. We're hoping this is a quick workaround for now, and we'd like to support this in the future

**Comment by miraodasilva** [comment]
I see, looking forward to this feature in the future. Thanks a lot for your time.

**Comment by vanpelt** [comment]
@miraodasilva I'm not positive I understand what you're asking, but our resuming feature should work for this if you want to "append" data to a run: https://docs.wandb.ai/guides/track/advanced/resuming

**Comment by sparisi** [comment]
Bump. 
It seems it is still not possible to overwite runs. I am in the same situation explained [here](https://github.com/wandb/wandb/issues/1450#issuecomment-722697783). 

I have a project with many runs on many datasets. 
I ran my code with ID datasetX-abc123 (abc123 actually depends on hyperparameter sweep). There was a bug for that specific dataset, so I had to rerun the sweep. However:
- If I delete the previous run with ID datasetX-abc123 I cannot re-run the same ID (as written in the `init()` doc and discussed in other issues. 
- If I don't delete and re-run with ID datasetX-abc123 then nothing is logged. Locally there are new files, but they are not synced. Even forcing sync doesn't do anything. 

I need to use datasetX-abc123 for consistency to analyze my hyperparameters, but I can't. 

Basically my only option is to delete the project if I want to reuse datasetX-abc123, but that means all my runs on other datasets will be lost. 

Please make it possible to either reuse deleted IDs, or to overwrite past runs.

**Comment by bio-mlhui** [comment]
The doc of arg "resume=None" for wandb.init is quite confusing. I thought wandb will overwrite a historic run starting counter 0. Turns out wandb will not change the historic values. 
![image](https://github.com/wandb/wandb/assets/107675879/b682f320-d386-4313-bd74-f182e8c82540)

**Comment by AndrewJBean** [comment]
This is really basic and obvious functionality that should be implemented. Can we get an update on this?

**Comment by nate-wandb** [comment]
Hi @AndrewJBean, I wanted to follow up and ask what you are seeing if you reuse a `run_id`? The behavior should be to automatically overwrite the data in the run. 

For example:
```
run = wandb.init(project='test', id="12345678")
run.log({
    "metric": 1,
})
```
Followed by 
```
run = wandb.init(project='test', id="12345678")
run.log({
    "metric": 2,
})
```
Results in the UI showing the value for `metric` as 2 and the data from the first run is lost. 

I believe this is what you are looking for correct?

**Comment by AndrewJBean** [comment]
Hi @nate-wandb , Here's an example where I would wish the existing logs to be overwritten:

```
[0]:wandb: WARNING (User provided step: 26500 is less than current step: 52501. Dropping entry: {'loss': ..., '_timestamp': 1701973708.0800202}).
```

deleting the existing run before rerunning the job in the UI leads to training job failure during startup (presumably the `wandb.init`)

**Comment by nate-wandb** [comment]
Hi @AndrewJBean, I see. Are you using the `resume` argument at all in your `wandb.init` ? The behavior you are getting sounds like you are resuming the run but if you don't pass the `resume` argument (or use `resume=None`) and pass the `id` then the behavior will be to overwrite the previous run and upload the new run in its place which is what it sounds like you are looking for correct?

**Comment by AndrewJBean** [comment]
Hi @nate-wandb , I tried out `resume=None` and it turns out i was misunderstanding the docs. I had understood it to mean that the whole run would be wiped and start from empty, but in reality what happens is that logs are overwritten per step value. While this is somewhat awkward, given that logs for future steps remain, it should be fine for my uses since both backfill and restore-from-checkpoint runs should fully overwrite those logs.
